tests/foreman/ui/test_ansible.py

```diff
@@ -467,22 +467,23 @@ def test_positive_non_admin_user_access_with_usergroup(
         :customerscenario: true
         """
         SELECTED_ROLE = 'RedHatInsights.insights-client'
+        name = gen_string('alpha')
         password = gen_string('alpha')
-        user = target_sat.api.User(
-            login=gen_string('alpha'), password=password, admin=False
-        ).create()
-        request.addfinalizer(user.delete)
+        user = target_sat.api.User(login=name, password=password, admin=False).create()
         user_gp = target_sat.api.UserGroup(
             name=gen_string('alpha'), user=[user], admin=True
         ).create()
         assert user.login in [u.read().login for u in user_gp.user]
-        with target_sat.ui_session(test_name, user.login, password) as session:
-            id = target_sat.nailgun_smart_proxy.id
-            target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
-            session.host_new.add_single_ansible_role(target_sat.hostname)
+        id = target_sat.nailgun_smart_proxy.id
+        target_sat.api.AnsibleRoles().sync(data={'proxy_id': id, 'role_names': [SELECTED_ROLE]})
+        with target_sat.ui_session(test_name, user=user.login, password=password) as session:
+            session.host_new.add_single_ansible_role(target_sat.hostname, SELECTED_ROLE)
             wait_for(lambda: session.browser.refresh(), timeout=5)
             ansible_roles_table = session.host_new.get_ansible_roles(target_sat.hostname)
             assert ansible_roles_table[0]['Name'] == SELECTED_ROLE
+        with target_sat.ui_session() as session:
+            session.user.delete(name)
+            assert not session.user.search(name)
```

Comment:
We can use api to delete the user or add finalizer to do it.

Reply:
If the teardown is causing conflict then I would suggest to try the following approach
```
session.host_new.remove_single_ansible_role(target_sat.hostname)
 session.login.logout()
```

and use teardown to delete the user.